### PR TITLE
Move isBackend logic to App\Mode

### PR DESCRIPTION
**Pull request by nupplaphil**
- Reduce App - dependency
- Added tests for "isBackend()" logic

### CODE REVIEW

File: src/App.php [hunk 1]
```diff
@@ -573,27 +561,17 @@ public function getUserAgent()
...
 	 *
 	 * @return bool Is it a known backend?
+	 *
+	 * @deprecated 2019.09 - use App\Mode->isBackend() instead
+	 * @see App\Mode::isBackend()
 	 */
 	public function isBackend()
 	{
-		return $this->isBackend;
+		return $this->mode->isBackend();
```
Comment (MrPetovan) [inline src/App.php]: Is there any way to retrieve the Mode object outside of the App class?

Comment (nupplaphil) [reply src/App.php]: If not directly per parameter, at least per `BaseObject::getClass(App\Mode::class)` :-)

Comment (nupplaphil) [reply src/App.php]: btw. the last occurence of `App->isBackend()` is here:
https://github.com/friendica/friendica/blob/afac4b7698166cfbc2550f863aa792a2bb7beb7a/src/Core/Worker.php#L1127

Comment (MrPetovan) [reply src/App.php]: Got it, please add this to the deprecation notice for future reference.

Comment (MrPetovan) [reply src/App.php]: Yes, it's the one that crashed my node last time.

Comment (MrPetovan) [reply src/App.php]: > If not directly per parameter, at least per `BaseObject::getClass(App\Mode::class)` :-)

Got it, please add this to the deprecation notice.

File: src/App.php [hunk 2]
```diff
@@ -573,27 +561,18 @@ public function getUserAgent()
...
-	}
-
 	/**
 	 * Returns true, if the call is from a backend node (f.e. from a worker)
 	 *
 	 * @return bool Is it a known backend?
+	 *
+	 * @deprecated 2019.09 - use App\Mode->isBackend() instead
+	 * @see App\Mode::isBackend()
+	 * Use BaseObject::getClass(App\Mode::class) to get these class directly per global registry
```
Comment (MrPetovan) [inline src/App.php]:
```suggestion
	 * Use BaseObject::getClass(App\Mode::class) to get the global instance of Mode
```

**Pull request merged**